### PR TITLE
Adds lava boats

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
One day I will get around to fixing lavaland core issues such as the map being bad and empty, but for now you get boats. (Though large swathes of impassable terrain with no sane way to cross is a big issue, so I guess this can be considered a band aid fix)

Boats are vehicles that can move 

From lava to land
From land to lava
From lava to lava
But not land to land

They will protect you from lava while you are on them.

TODO: 

I need sprites for the boat and for an oar. 

I will probably add a way to craft this out of goliath hide.

:cl: Kor
rscadd: You can now craft boats and oars out of goliath hides. These boats can move in lava.
rscadd: Added a spooky, upgraded boat to necropolis chests.
/:cl:

### CODE REVIEW

**Comment by CosmicScientist** [comment]
> >thumbs down

**Comment by GunHog** [comment]
So, miners are a bunch of...high tech nomads?

**Comment by tedward1337** [comment]
Migrants, as it were gunhog

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 1]
```diff
@@ -334,3 +331,33 @@
...
+		..()
+	else
+		user << "Boats don't go on land!"
+		return 0
+
+/obj/item/weapon/oar
+	name = "oar"
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "keysec"
+	desc = "Not to be confused with the kind Research hassles you for."
```
Comment (Cheridan) [inline code/modules/mining/lavaland/necropolis_chests.dm]: THIS PUN IS CUTE

**Comment by KorPhaeron** [comment]
@GunHog 

There is no rhyme or reason to this, someone asked for boats in #coderbus so I made boats

Lavaland is transitioning into the standard SS13 development phase where we've run out of ideas and just add gimmicks instead (I will try and have an actual Design Plan written up Soon(TM))

**Comment by GunHog** [comment]
>where I've run out of ideas and just add gimmicks instead

Focus on Chaplain armor, then! :D

**Comment by TankNut** [comment]
I'd say keep the mining cart sprite, because seeing a miner traverse a river of lava in a wooden cart seems hilarious.

**Comment by KorPhaeron** [comment]
But I nees the mining cart sprite for when I add minecarts

**Comment by GunHog** [comment]
>But I nees the mining cart sprite for when I add minecarts

Can they at least be _Hovercarts_? Should at least be somewhat space-tech themed.

**Comment by CosmicScientist** [comment]
>Can they at least be Hovercarts? Should at least be somewhat space-tech themed.
Maaaglev.

**Comment by KorPhaeron** [comment]
>he thinks lavaland is space themed

But yeah maglevs sound way cooler (and easier to sprite smoothing rails for) than minetracks. I'll make a PR for those as well.

**Comment by FantasticFwoosh** [comment]
Indeed, ash walkers shall row war canoes down the lava river to the mining station, as a unintended buff to raiding parties. (depending on the means of attainment)

How physically tough would lava boats be? Because more often than not, if it deconstructs on monster collision like a crate, it'll drop its crew directly into liquid fire which is pretty much a lose condition and the boat will be unreachable without another.

**Comment by swankcookie** [comment]
I'm working on sprites right now, and I suggested this because I'm trying to work towards the gulag not being a "GG, I AFK scenario"

**Comment by CosmicScientist** [comment]
Can't wait to sail in that dragon boat.

**Pull request merged**